### PR TITLE
Small footer rework

### PR DESCRIPTION
**Pull request by davidfischer**
This reworks the footer specifically to:

- Add an email newsletter signup form. Back in 2014 ([see WayBack Machine](https://web.archive.org/web/20140415000000/http://readthedocs.org:80/)) we had an email newsletter in the footer. It got way too much spam signups and would have cost us money to continue to use it. Our email newsletter is now "double opt-in" meaning not only do users have to type in an email address but they have to confirm their email address by receiving an email and clicking a confirm to sign up to the list. This should drastically reduce the spam signups and increase quality overall.
- Balances the 3 footer columns a bit better
- Centers the social media/github icons
- The version number is on its own line. Previously this wrapped slightly awkwardly.

From a browser perspective, I tested this down to IE11.

## SCREENSHOT
screen shot 2018-05-25 at 4 34 04 pm

### CODE REVIEW

**Comment by davidfischer** [comment]
The CI errors are unrelated.

**Comment by ericholscher** [comment]
I like this in general, a few comments:

* It feels like the blog & newsletter should tie together more. They are effectively the same content, but here it feels like they are separate things
* I'm not in love with centering the social icons. It makes it feel like a heading or something, instead of a section. Do we need icons for this?
* I'm not sure about the "Read the Docs", "About Us" , and "Email Newsletter" headings. They feel weirdly disjoint in scope. 

A proposal:

```
Stay Updated
------------

Blog
Twitter
Newsletter Sub Form

Get Involved
------------

Getting Started
Docs
Contributing
GitHub
Team

Business
--------

Advertise with Us
Gold Supporters
Donate 
Privacy

```

**Comment by davidfischer** [comment]
Debatably "Team" should go under "Business Info". Also possibly "Business Info" should be simply "Business".

File: readthedocs/templates/base.html [hunk 1]
```diff
@@ -105,76 +105,97 @@
...
+              <div>
+                <h4>{% trans 'Stay Updated' %}</h4>
+
+                <ul>
+                  <li>
+                    <a href="https://blog.readthedocs.com/">{% trans 'Blog' %}</a>
+                  </li>
+                </ul>
+
+                <p>{% trans 'Sign up for our newsletter to get our latest blog updates delivered to your inbox weekly.' %}</p>
```
Comment (ericholscher) [inline readthedocs/templates/base.html]: I might also link the `blog` here to the blog, and perhaps even more it from above. That might make it a bit more seemless.

**Pull request merged**